client/basic-authorship/src/basic_authorship.rs

```diff
@@ -358,8 +358,16 @@ where
 			);
 		});
 
+		let block_size_limit = block_size_limit.unwrap_or(self.default_block_size_limit);
+
+		let ensure_block_limit = if self.include_proof_in_block_size_estimation {
+			Some((true, block_size_limit))
+		} else {
+			None
+		};
+
 		for inherent in inherents {
-			match block_builder.push(inherent) {
+			match block_builder.push(inherent, ensure_block_limit) {
```

Comment:
WE should only pass `None` here, inherents are required/should be required.

Reply:
done